### DISCUSSION TITLE
Encode DoVi AV1 + HDR10+

### DISCUSSION THREAD

**Discussion by llutic** [gianni-rosato/svt-av1-psy#47] [Q&A] [answered]
Hello,

I tried to encode in AV1 with DoVi and it worked perfectly.

Now I'm trying to encode a source that has DoVi (in profile 8.1 in this case) and HDR10+ but in this case I lose the hdr10+ layer for only simple HDR10. I've also tried encoding seuelemnt with HDR10+ without DoVi and again the encoding doesn't keep the dynamic layer.

I've seen that you can encode HDR10+ in AV1 via Rigaya hardware encoders with NVENC for example, but I don't own any AV1 hardware encoders and besides I don't want to encode with my library of films and series.

Here are the parameters I use via staxrip (with SVT-AV1-PSY 2.0.0-7+6-62325458-.Mod-by-Patman.-x64-msvc1937) :

--rc 0 --crf 27 --progress 3 --enable-hdr 1 --preset 7 --tune 0 --keyint 2s --scd 1 --enable-overlays 1 --color-primaries 9 --transfer-characteristics 16 --matrix-coefficients 9 --chroma-sample-position 2 --mastering-display "G(0.2650,0.6900)B(0.1500,0.0600)R(0.6800,0.3200)WP(0.3127,0.3290)L(1000.0000,0.0050)" --content-light "699,333"

Am I missing something, or does SVT-AV1 still not support HDR10+ encoding?

Thanks!

**Comment by gianni-rosato** [comment, marked as answer]
Hi,

Thank you for your detailed question & for providing the encoding parameters you are using. SVT-AV1-PSY currently supports encoding with Dolby Vision (DoVi) using the `--dolby-vision-rpu` parameter, given the encoder is built with the `enable-libdovi` flag passed to the build script on Linux & macOS. However, SVT-AV1-PSY does not natively support HDR10+ encoding at this time.

**Comment by llutic** [comment]
Hello,

Thank you for your reply!

Is this a feature that is planned or will we have to wait for SVT-AV1 to support it so that SVT-AV1-PSY has it or can we expect like dolby vision and that it is based on quietdovi work on HDR10+?

**Comment by gianni-rosato** [comment]
I'll look into it if there is enough demand. It seems compelling to support both HDR10+ & Dolby Vision encoding

**Comment by llutic** [comment]
Thank you !  I hope this can be done :)

**Comment by quietvoid** [comment]
See https://github.com/quietvoid/SVT-AV1/commit/18f8213f8bdfc3f1d5728dda31f33b9e41ed5688
Haven't tested it yet.

**Reply by gianni-rosato** [reply]
Do you wanna open a PR to our testing branch? We'd love to have you contribute!

**Reply by quietvoid** [reply]
Maybe when it's tested.

**Reply by llutic** [reply]
Hello,

I was able to test SVT-AV1-PSY compliant from the branch modified by quietvoid

![Capture d'écran 2024-05-30 0209482](https://github.com/gianni-rosato/svt-av1-psy/assets/113514011/eabecac7-034e-4712-be59-434c2310d00c)

My test files are here (1 DV, 1 HDR10+ and 1 DV + HDR10+) : https://drive.google.com/drive/folders/16fz7j2JVv7h4tOcgJkXofUq9z8Fb31TO?usp=sharing

I can play Dolby Vision correctly with Kodi version 21 on Android - Google TV 11 (TCL 55C825), but the version with only HDR10+ on my TV only decodes the HDR part. After testing with MPV on PC Win 11, it seems to detect HDR10+ correctly.

![Capture d'écran 2024-05-30 020924](https://github.com/gianni-rosato/svt-av1-psy/assets/113514011/e4d9769e-b03e-443d-97bd-94fff1442212)

My TV has a hardware codec to decode HDR10+ AV1 (OMX.MS.AV1.Decoder with profile AV1ProfileMain10HDR10 (0x1000): AV1Level51 (0x2000) and AV1ProfileMain10HDR10Plus (0x2000): AV1Level51 (0x2000))

so I don't think the software is tracking the hardware on my TV yet

If there's anything else I can do, please don't hesitate :)

**Reply by quietvoid** [reply]
I've tested it on my FireTV Stick 4K Max using Kodi and it played in HDR10+ on my TV, which supports both HDR10+ and Dolby Vision.
TCL Q5 (43Q550G)

**Comment by llutic** [comment]
I've just analised a file I encoded by modifying a few parameters and the good news is that HDR10+ is detected in the stream :)

![image](https://github.com/gianni-rosato/svt-av1-psy/assets/113514011/f6e93e59-bc5f-44d3-9c87-fc40c092b25e)

Apparently the info isn't displayed in mediainfo, so I opened an issue on its github: https://github.com/MediaArea/MediaInfo/issues/862

but for the moment no news

would it be possible to include this modification in the main branch and also add it to the windows version and not just linux?

Thanks :)
